sources/packages/backend/apps/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -212,6 +235,47 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
     return found.length === locations.length;
   }
 
+  /**
+   * Check if location code is already registered for the institution.
+   * @param locationCode location code.
+   * @param locationId location id in case it is an update of the location.
+   * @param institutionId institution id in case it is a new location.
+   * @returns true in case there is already a location code for the institution.
+   */
+  async hasLocationCodeForInstitution(
+    locationCode: string,
+    locationId?: number,
```

Comment:
For these cases, we have adopted the `options` to make it clear from the method consumer perspective, as shown below.
```ts
// Declared as 
async hasLocationCodeForInstitution(
    locationCode: string,
    options: {
        locationId?: number,
        institutionId?: number
    }
```
```ts
// From consumer
something.hasLocationCodeForInstitution("aaaa", {locationId: 123});
```

Reply:
Ok. Should I use the options or change the api to always pass the institutionId?

Reply:
ok! Done with options object. Thanks!